src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaController.java

```diff
@@ -253,6 +261,51 @@ private void readRcaEnabledFromConf() {
             rcaEnabled = rcaEnabledDefaultValue;
           }
         });
+
+    // If RCA is enabled, update Analysis graph with Muted RCAs value
+    if (rcaEnabled) {
+      LOG.debug("Updating Analysis Graph with Muted RCAs");
+      readAndUpdateMutesRcas();
+    }
+  }
+
+  /**
+   * Reads the mutedRCAList value from the rca.conf file, performs validation on the param value
+   * provided and on successful validation, updates the AnalysisGraph with muted RCA value.
+   *
+   * <p>In case all the RCAs in param value are incorrect, return without any update.
+   */
+  private void readAndUpdateMutesRcas() {
+    // If the rca config file has been updated since the lastModifiedTimeInMillisInMemory in memory,
+    // refresh the `muted-rcas` value from rca config file.
+    long lastModifiedTimeInMillisOnDisk = rcaConf.getLastModifiedTime();
+    if (lastModifiedTimeInMillisOnDisk > lastModifiedTimeInMillisInMemory) {
+      try {
+        Set<String> rcasForMute = new HashSet<>(rcaConf.getMutedRcaList());
+        LOG.info("RCAs provided for muting : {}", rcasForMute);
+
+        Set<String> graphNodeNames = new HashSet<>();
+        RcaUtil.getAnalysisGraphComponents(rcaConf).forEach(
```

Comment:
This is not the right method to call. This will re-initialize the entire Analysis graph.

Reply:
I was thinking we can do something much simpler. As [here](https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/pull/167/commits/54bc23619d9c66643fcea6afb74fa2f188ab4a94#diff-c1b8be1a0f84a6a0b5f51bb1041f9c9bL34)  we are iterating over all nodes, we also create another HashSet<String> that contains the node Names only. Then the method [here](https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/pull/167/commits/54bc23619d9c66643fcea6afb74fa2f188ab4a94#diff-c1b8be1a0f84a6a0b5f51bb1041f9c9bR116) serves as a getter for it. This avoids traversing all the nodes twice and also re-creating the HashSet on each call.